st2common/st2common/util/action_db.py

```diff
@@ -213,14 +218,14 @@ def get_args(action_parameters, action_db):
     positional_args = []
     positional_args_keys = set()
     for _, arg in six.iteritems(position_args_dict):
-        positional_args.append(str(action_parameters.get(arg)))
+        positional_args.append(_quote_text_lines(str(action_parameters.get(arg))))
```

Comment:
Edit: I've looked at the changes again.

I don't think this is the right place to escape the arguments. Action utils are (and should be) unaware of where the action runs and escaping is different on Windows and Linux. This mean escaping should happen in a different (right) layer - ideally this would be SSH runner or system actions used by the SSH runner.

I would personally prefer to do it in the SSH runner (closest to the place where commands are actually executed) and I tried to convince @lakshmi-kannan to do that in the other PR, but he did it in the system models...